_health-care/_js/_components/date_input.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import _ from 'lodash';
+
+class DateInput extends React.Component {
+  constructor() {
+    super();
+    this.state = {month: '', date: '', year: ''};
+    this.handleMonthChange = this.handleMonthChange.bind(this);
+    this.handleDateChange = this.handleDateChange.bind(this);
+    this.handleYearChange = this.handleYearChange.bind(this);
+  }
+
+  componentWillMount() {
+    this.id = _.uniqueId('date-input-');
```

Comment:
Is this referencing an array and then setting a unique ID prefixed with `date-input-`? And then where is `componentWillMount` called? Is that something that is automatically called by React? I see it's working when I run it locally, just want to be sure I understand how!